app/src/main/java/org/dhis2/usescases/programEventDetail/ProgramEventMapper.kt

```diff
@@ -51,10 +53,98 @@ class ProgramEventMapper @Inject constructor(val d2: D2) {
             event.status()!!,
             hasExpired || !inOrgUnitRange,
             attrOptCombo,
-            event.geometry()
+            event.geometry(),
+            isEventEditable(event.uid())
         )
     }
 
+    fun isEventEditable(eventUid: String): Boolean {
+        val event =
```

Comment:
can you extract this to a class? so is ready to move for sdk.. :dancers:

Reply:
I was thinking about it. I'll do it yes :)